drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -524,312 +500,327 @@
+
+          Those are all the header fields defined for a E2E Option-Type.</t>
+
+          <t>A document defining a new IOAM Integrity Protected Option-Type that
+          intends to use the Integrity Protection Method defined in this
+          document MUST also specify a list of corresponding Option-Type header
+          fields that participate in the integrity protection of
+          IOAM-Data-Fields, as above. The same logic can be applied to select
+          header fields, by following this simple rule:
+            <list style="bullets">
+              <t>Only immutable IOAM Option-Type header fields that provide
+              context or meaning to IOAM-Data-Fields are considered, others are
+              excluded (e.g., mutable or reserved fields). For example, the
+              Namespace-ID, which is common to all IOAM Option-Types, MUST
+              always be included in the list. Once specified, the list MUST NOT
+              change for interoperability reasons.</t>
+            </list>
           </t>
+        </section>
+      </section>
 
-          <t>The decapsulating node MUST act as a Validator. The decapsulating
-          node MUST NOT generate a signature based on IOAM-Data-Fields it has
-          inserted, if any, and therefore MUST NOT update the Signature field.</t>
-        </list>
+      <section anchor="IPM-Transit" title="Transit node">
+        <t>For a transit node, the Integrity Check Value (ICV) is the result of
+        a GMAC operation over the received ICV field and immutable
+        IOAM-Data-Fields added by the transit node. In other words, with the
+        received Nonce field provided to GMAC, the transit node performs the
+        following operation:
+          <list style="bullets">
+            <t>AAD = (ICV field || node's immutable IOAM-Data-Fields)</t>
+            <t>ICV = GMAC(AAD)</t>
+          </list>
         </t>
-      </list>
-      </t>
 
-      <t>The method assumes that symmetric keys have been distributed to
-      the respective nodes as well as the Validator(s). The details of the
-      mechanisms used for key distribution are outside the scope of this
-      document.</t>
+        <t>The transit node updates the ICV field in the IOAM Integrity
+        Protection header. Reminder: it MUST always be a 16-octet ICV.</t>
+
+        <t>If the transit node does not add any IOAM-Data-Fields (e.g., it
+        only modifies mutable IOAM-Data-Fields or does nothing), then the
+        transit node MUST NOT update the ICV field in the IOAM Integrity
+        Protection header.</t>
+      </section>
+
+      <section anchor="IPM-Decap" title="Decapsulating node">
+        <t>In the context of integrity protection, the decapsulating node can
+        either act as a transit node (see <xref target="IPM-Transit"/> above) or
+        directly as the Validator (see <xref target="IPM-Validator"/> below).
+        The former is an alternative to put the Validator out of the forwarding
```

Comment:
IMHO this could be confusing. You make a difference between "IOAM transit nodes" and "IOAM-IPM transit nodes".
Just calling a decap node a transit node could give raise to mis-interpretations. IMHO we should either use "IOAM transit nodes, IOAM decap nodes" and "IOAM-IPM transit nodes" etc. throughout the doc or re-phrase this paragraph.

E.g., you could say: The Validator role can either be performed by the IOAM decapsulating node, or another node.

Question: If the Validator is not the IOAM decap node, how would this work, given that the decap node removes the IOAM header?

Reply:
> IMHO this could be confusing. You make a difference between "IOAM transit nodes" and "IOAM-IPM transit nodes". Just calling a decap node a transit node could give raise to mis-interpretations. IMHO we should either use "IOAM transit nodes, IOAM decap nodes" and "IOAM-IPM transit nodes" etc. throughout the doc or re-phrase this paragraph.

+1 to re-phrase the paragraph (I prefer that vs the other solution).

> E.g., you could say: The Validator role can either be performed by the IOAM decapsulating node, or another node.

Not sure how to keep a sub-section for the decapsulating node in that case. I'll think about it tonight and come with proposed text tomorrow.

> Question: If the Validator is not the IOAM decap node, how would this work, given that the decap node removes the IOAM header?

Most obvious idea I have in mind: a packet copy is sent by the decapsulating node (before the IOAM removal) to the Validator. There are other possibilities, but I think we should stay away from this. Maybe add a sentence to say that the decapsulating node would need to send IOAM data to the Validator, but is out of scope?

Reply:
> Maybe add a sentence to say that the decapsulating node would need to send IOAM data to the Validator, but is out of scope?

That could work if you're really keen to keep that option. We could say something like - "If the decapsulating node does not perform the function of the Validator, the decapsulating node needs to send the IOAM related fields to the Validator. The method how the IOAM data is sent to the Validator is out of scope for this document".

Reply:
+1, thanks! Indeed, I think it'd be good to have both alternatives, in case decap==Validator is too heavy to deploy depending on situations.